### PR TITLE
Filter by usernote tag

### PR DESCRIPTION
**Pull request by dequeues**
Implement #534 

There is now a dropdown on the usernote search page of all the user note titles, and "All" which you can filter the usertags with.
First PR here, please be kind. 👋

### CODE REVIEW

File: extension/data/modules/usernotes.js [hunk 1]
```diff
@@ -1104,7 +1109,7 @@ function usernotes () {
                     }
 
                     return true;
-                });
+                }).filter(user => user.notes.filter(notes => filterKey === 'all' || notes.type === filterKey).length > 0);
```
Comment (eritbh) [inline extension/data/modules/usernotes.js:1112]: What you have here works perfectly fine, we can probably get better performance out of it by putting this in the existing `notes` filter that starts on line 1099. Something like this:

```js
const filteredData = allUsers.map(user => ({
    name: user.name,
    notes: user.notes.filter(note => {
        // Filter out notes not matching `contentText`
        if (!note.note.toLowerCase().includes(contentText.toLowerCase())) {
            return false;
        }

        // Filter out note types that are not `filterKey`
        if (filterKey !== 'all' && filterKey !== note.type) {
            return false;
        }

        return true;
    }),
})).filter(user => // ...
```

Doing all the filtering in the same `.filter` call means we don't need to iterate over all the users and notes multiple times, which for usernotes can be a big concern (some subreddits have a *lot* of usernotes). This new structure will be a little more flexible and allows us to filter on multiple different properties within a single loop.

This is very minor, but I also changed the iteration variable from `notes` to `note` - inside the loop, it refers to a single note object, not an array of them, so the singular name makes the code just a bit easier to read.

File: extension/data/modules/usernotes.js [hunk 2]
```diff
@@ -1045,6 +1045,10 @@ function usernotes () {
                     <div class="tb-un-info">
                         <span class="tb-info">There are ${userCount} users with ${noteCount} notes.</span>
                         <br> <input id="tb-unote-user-search" type="text" class="tb-input" placeholder="search for user"> <input id="tb-unote-contents-search" type="text" class="tb-input" placeholder="search for note contents">
+                        <select name="tb-un-filter" id="tb-un-filter" class="selector tb-action-button">
+                            <option value="all" default>All</option>
+                            ${colors.map(op => `<option value=${op.key}>${op.text}</option>`).join('')}
```
Comment (eritbh) [inline extension/data/modules/usernotes.js:1050]: We unfortunately aren't super consistent about this in the rest of the code right now, but any time user input is put into HTML, it should be run through `TBHelpers.htmlEncode()` first.

```suggestion
                            ${colors.map(op => `<option value=${op.key}>${TBHelpers.htmlEncode(op.text)}</option>`).join('')}
```

**Review by eritbh — CHANGES_REQUESTED** [review]
Hey, thanks so much for taking the time to write a PR for this! I just have a couple suggestions before we can merge it.

**Comment by dequeues** [comment]
Agree with both - I'll update the PR soon. Thanks.

**Review by eritbh — APPROVED** [review]

**Comment by eritbh** [comment]
Thanks again for the contribution! This should land in our next feature update.

**Pull request merged**